proposals/ideas/xxx-padding-machines-onions.txt

```diff
@@ -0,0 +1,302 @@
+   a histogram which specifies the parameters described in the paragraphs
+   above. After all the padding finishes, it moves to END state.
+
+   We also set a special WTF-PAD flag which keeps the circuit open even after
+   the introduction is performed, and until all the padding has been sent
+   out. In particular, the origin-side machine needs in average 3.3 secs to
+   deplete all its padding cells, and hence it will slightly increase the
+   circuit lifetime. See section [IMPROVEMENTS] for a further low-hanging-fruit
+   improvement in this direction.
+
+5.2. Client-side rendezvous circuit hiding machines
+
+   The rendezvous circuit machines apply on client-side rendezvous circuits and
+   only after the rendezvous point has been established (REND_ESTABLISHED has
+   been received). Up to that point, the following cell sequence has been
+   observed on the circuit:
+
+    [EXTEND2] -> EXTENDED2 -> [EXTEND2] -> EXTENDED2 -> [ESTABLISH_REND] -> REND_ESTABLISHED
+
+   which matches the general circuit construction sequence [CIRCCONSTRUCTION]
+   up to the first 6 cells. However after that, normal rendezvous circuits
+   receive a RENDEZVOUS2 cell followed by a [BEGIN] and a CONNECTED, which does
+   not fit the circuit construction sequence we are trying to imitate.
+
+   Hence our machine gets activated right after REND_ESTABLISHED is received,
+   and continues by sending a [PADDING_NEGOTIATE] and a [DROP] cell, before
+   receiving a PADDING_NEGOTIATED and a DROP cell, effectively blending into
+   the general circuit construction sequence on the first 10 cells.
+
+   After that our machine gets deactivated, and we let the actual rendezvous
+   circuit shape the traffic flow. Since rendezvous circuits usually immitate
+   general circuits (their purpose is to surf the web), we can expect that they
+   will look alike.
+
+   In terms of overhead, this machine is quite light. Both sides send 2 padding
+   cells, for a total of 4 padding cells.
+
+6. Overhead analysis
+
+   Given the parameters above, intro circuit machines have an overhead of 11
+   padding cells, and rendezvous circuit machines have an overhead of 4
+   cpadding ells.  . This means that for every intro and rendezvous circuit
+   there will be an overhead of 15 padding cells in average, which is about
+   7.5kb.
+
+   In the PrivCount paper [1] we learn that the Tor network sees about 12
+   million successful descriptor fetches per day. We can use this figure to
+   assume that the Tor network also sees about 12 million intro and rendezvous
+   circuits per day. Given the 7.5kb overhead of each of these circuits, we get
+   that our padding machines infer an additional 94GB overhead per day on the
+   network, which is about 3.9GB per hour.
+
+   XXX Isn't this kinda intense????? Using the graphs from metrics we see that
+       the Tor network has total capacity of 300 Gbit/s which is about 135000GB per
+       hour, so 3.9GB per hour is not that much, but still...
```

Comment:
Nah, this is fine :fire:.

We need to skate to where the puck is going to be, or we'll deadlock forever on stuff like this and never deploy anything. The plan is to implement proposal #265 and ask people to run more middle relays, at which point 3.9GB per hour will become negligible.

Reply:
ACK